common/models/transducer/alignment.py

```diff
@@ -64,3 +65,13 @@ def rnnt_alignment(source, **kwargs):
   _, alignment = tf_forward_shifted_rnnt(log_probs, targets.get_placeholder_as_batch_major(), enc_lens, dec_lens,
                                          blank_index=blank_idx, debug=False, with_alignment=True)
   return alignment  # (B, T)
+
+
+def rna_alignment_out_type(sources, **_kwargs) -> Data:
+  return Data(name="rna_alignment_output", sparse=True, dim=sources[0].output.dim,
+              size_placeholder={0: sources[2].output.size_placeholder[0]})
+
+
+def rnnt_alignment_out_type(sources, **_kwargs) -> Data:
+  return Data(name="rnnt_alignment_output", sparse=True, dim=sources[0].output.dim,
+              size_placeholder={0: sources[1].output.size_placeholder[0] + sources[2].output.size_placeholder[0]})
```

Comment:
Did you copy this from somewhere?
This is not 100% correct behavior because now new tensors should be created at this point.
(E.g. see the discussion [here](https://github.com/rwth-i6/returnn/issues/391) about `size_placeholder`.)
But maybe this is not too much an issue here.

Reply:
Took it from old configs such as [here](https://github.com/rwth-i6/returnn-experiments/blob/68595c0b6ad4999b9fcfc9bbbc31f589f58f6e78/2021-transducer-thesis-merboldt/librispeech/rnnt-fs.bpe1k.readout.zoneout.lm-embed256.lr1e_3.no-curric.bs12k.mgpu.retrain1.common.config.py):

```python
  subnet["fullsum_alignment"] = {
          "class": "eval",
          "from": ["output_log_prob", "base:data:" + _target, "base:encoder"],
          "eval": rna_fullsum_alignment,
          "out_type": lambda sources, **kwargs: Data(name="rna_alignment_output", sparse=True, dim=_targetb_num_labels,
                                                     size_placeholder={0: sources[2].output.size_placeholder[0]}),
          "is_output_layer": True
      }
```

Cant see from [here](https://github.com/rwth-i6/returnn/issues/391) how to properly do it.

Reply:
> Took it from old configs such as [here](https://github.com/rwth-i6/returnn-experiments/blob/68595c0b6ad4999b9fcfc9bbbc31f589f58f6e78/2021-transducer-thesis-merboldt/librispeech/rnnt-fs.bpe1k.readout.zoneout.lm-embed256.lr1e_3.no-curric.bs12k.mgpu.retrain1.common.config.py):
> 
> ```python
>   subnet["fullsum_alignment"] = {
>           "class": "eval",
>           "from": ["output_log_prob", "base:data:" + _target, "base:encoder"],
>           "eval": rna_fullsum_alignment,
>           "out_type": lambda sources, **kwargs: Data(name="rna_alignment_output", sparse=True, dim=_targetb_num_labels,
>                                                      size_placeholder={0: sources[2].output.size_placeholder[0]}),
>           "is_output_layer": True
>       }
> ```

But I'm not speaking about this. This code does not create new tensors.

I'm speaking e.g. about `rnnt_alignment_out_type`.
 
> Cant see from [here](https://github.com/rwth-i6/returnn/issues/391) how to properly do it.

There is no discussion on how to do it. It just explains the issue.